src/ProtocolV3TestBase.sol

```diff
@@ -392,6 +408,82 @@ contract ProtocolV3TestBase is CommonTestBase {
     return IERC20(config.aToken).totalSupply() + supplyAmount > (config.supplyCap * 10 ** config.decimals);
   }
 
+  function _e2eTestLiquidation(
+    IPool pool,
+    address user,
+    address liquidator,
+    ReserveConfig memory collateralConfig,
+    ReserveConfig memory testAssetConfig,
+    uint256 amount
+  ) internal {
+    uint256 collateralAmountBefore = IERC20(collateralConfig.aToken).balanceOf(user);
+    this._borrow(testAssetConfig, pool, user, amount, false);
+    assertGt(IERC20(testAssetConfig.variableDebtToken).balanceOf(user), 0, 'DEBT_ZERO');
+    IPoolConfigurator configurator = IPoolConfigurator(
+      IPoolAddressesProvider(pool.ADDRESSES_PROVIDER()).getPoolConfigurator()
+    );
+    // set ltv/lt to 1bps which enables liquidation on the position
+    vm.prank(IPoolAddressesProvider(pool.ADDRESSES_PROVIDER()).getACLAdmin());
+    configurator.configureReserveAsCollateral(
+      collateralConfig.underlying,
+      1,
+      1,
+      10001
+    );
+    _liquidate(collateralConfig, testAssetConfig, pool, liquidator, user, amount, true);
+    // hf should be low enough to have the entire position be liquidated
+    assertEq(IERC20(testAssetConfig.variableDebtToken).balanceOf(user), 0, 'LIQUIDATION_DEBT_NOT_ZERO');
```

Comment:
Can move this assertion into `_liquidate`

Reply:
It actually depends on if the HF is below 95% or not, so for here it makes more sense to be outside imo.